src/controllers/expenses.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-fallthrough */
+'use strict';
+
+const {
+  getAll,
+  getById,
+  create,
+  remove,
+  update,
+  getAllForUser,
+  getAllForCategory,
+  getAllBetweenDates,
+} = require('../services/expenses');
+
+const { getById: getUserById } = require('../services/users');
+
+const getAllExpenses = (req, res) => {
+  let expenses = getAll();
+  const { userId } = req.query;
+  const { categories } = req.query;
+  const { from, to } = req.query;
+
+  switch (true) {
+    case !!userId:
+      if (userId) {
+        expenses = getAllForUser(userId);
+      }
+
+    case !!categories:
+      if (categories) {
+        expenses = getAllForCategory(expenses, categories);
+      }
+
+    case (!!from && !!to):
+      if (from && to) {
+        const timestampFrom = new Date(from).getTime();
+        const timestampTo = new Date(to).getTime();
+
+        expenses = getAllBetweenDates(expenses, timestampFrom, timestampTo);
+      }
+  }
+
+  res.send(expenses);
+};
+
+const getExpenseById = (req, res) => {
+  const { id } = req.params;
+  const foundExpense = getById(id);
+
+  if (!foundExpense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(foundExpense);
+};
+
+const createExpense = (req, res) => {
+  const user = getUserById(req.body.userId);
+
+  if (!user) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  if (Object.keys(req.body) === 0) {
```

Comment:
Use 'Object.keys(req.body).length === 0' to check if the req.body is empty.

Reply:
Done